torchvision/datasets/_stereo_matching.py

```diff
@@ -261,7 +268,7 @@ def _read_disparity(self, file_path: str) -> Tuple:
         valid_mask = None
         return disparity_map, valid_mask
 
-    def __getitem__(self, index: int) -> Tuple:
+    def __getitem__(self, index: int) -> Union[T1, T2]:
```

Comment:
Output is always a 4-tuple here, right?

```suggestion
    def __getitem__(self, index: int) -> T1:
```

Reply:
Yes that's what I did first. But the problem is that we return the inherited function and so mypy cannot reconcile this with the signature of the inherited function :/

More concretely:
- if we type T1 the output signature of the overloaded method
- what is returned is the inherited magic method with singature `Union[T1, T2]` and even though it's actually `T1` because of the submethod overload, mypy doesn't pick it up :/

Reply:
```
torchvision/datasets/_stereo_matching.py:284: error: Incompatible return value type (got
"Union[Tuple[Any, Any, Optional[ndarray[Any, Any]], ndarray[Any, Any]], Tuple[Any, Any, Optional[ndarray[Any, Any]]]]",
expected "Tuple[Any, Any, Optional[ndarray[Any, Any]], ndarray[Any, Any]]")  [return-value]
            return super().__getitem__(index)
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

How would you see that going with `cast` I'm confused?